### PR TITLE
add Make payment and Submit details steps in the docs

### PR DESCRIPTION
**Pull request by lojzatran**

### CODE REVIEW

File: extension/docs/WebComponentsIntegrationGuide.md [hunk 1]
```diff
@@ -144,3 +144,179 @@ Pass the `origin key` to your front end. You might use this origin key to be abl
 
 > Note: First 2 steps are optional if origin key and payment methods has been already cached by the merchant server.
  
+## Step 3: Make a payment
+After the shopper submits their payment details or chooses to pay with a payment method that requires a redirection,
+the Adyen Web Components will generate a `makePaymentRequest`. Consult [Adyen documentation](https://docs.adyen.com/api-explorer/#/PaymentSetupAndVerificationService/payments) to see which parameters 
+are necessary for the current payment request.
+
+[Update CTP payment](https://docs.commercetools.com/http-api-projects-payments#update-payment) with `makePaymentRequest` custom field. 
+```json
```
Comment (ahmetoz) [inline extension/docs/WebComponentsIntegrationGuide.md:153]: can you also add the example request as json before showing it in, check the step 2, it's not easy to read and understand a stringified JSON.

File: extension/docs/WebComponentsIntegrationGuide.md [hunk 2]
```diff
@@ -144,3 +144,179 @@ Pass the `origin key` to your front end. You might use this origin key to be abl
...
+A CTP payment with `makePaymentResponse` for the case where user has to be redirected to a payment provider page for further authentication:
+```json
+{
+  "custom": {
+    "type": {
+      "typeId": "type",
+      "key": "ctp-adyen-integration-web-components-payment-type"
+    },
+    "fields": {
+      "makePaymentRequest": "{\"riskData\":{\"clientData\":\"xxx\"},\"paymentMethod\":{\"type\":\"scheme\",\"holderName\":\"Lam Tran\",\"encryptedCardNumber\":\"adyenjs_...\",\"encryptedExpiryMonth\":\"adyenjs_...\",\"encryptedExpiryYear\":\"adyenjs_...\",\"encryptedSecurityCode\":\"adyenjs_...\"},\"billingAddress\":{\"street\":\"Test Strasse\",\"houseNumberOrName\":\"10\",\"postalCode\":\"80000\",\"city\":\"Munich\",\"stateOrProvince\":\"Bayern\",\"country\":\"DE\"},\"browserInfo\":{\"acceptHeader\":\"*/*\",\"colorDepth\":24,\"language\":\"en-US\",\"javaEnabled\":false,\"screenHeight\":1050,\"screenWidth\":1680,\"userAgent\":\"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.129 Safari/537.36\",\"timeZoneOffset\":-120},\"reference\":773,\"amount\":{\"currency\":\"EUR\",\"value\":1000}}",
```
Comment (ahmetoz) [inline extension/docs/WebComponentsIntegrationGuide.md:178]: Can we use the examples from Adyen documentation? It's better than this one.

File: extension/docs/WebComponentsIntegrationGuide.md [hunk 3]
```diff
@@ -144,3 +144,179 @@ Pass the `origin key` to your front end. You might use this origin key to be abl
...
  
+## Step 3: Make a payment
+After the shopper submits their payment details or chooses to pay with a payment method that requires a redirection,
+the Adyen Web Components will generate a `makePaymentRequest`. Consult [Adyen documentation](https://docs.adyen.com/api-explorer/#/PaymentSetupAndVerificationService/payments) to see which parameters 
+are necessary for the current payment request.
+
+[Update CTP payment](https://docs.commercetools.com/http-api-projects-payments#update-payment) with `makePaymentRequest` custom field. 
+```json
+{
+  "version": "PAYMENT_VERSION",
```
Comment (ahmetoz) [inline extension/docs/WebComponentsIntegrationGuide.md:155]: Can we use the examples from Adyen documentation? It's better than this one. https://docs.adyen.com/checkout/components-web#step-3-make-a-payment Also it's easier to follow if they are the same. Also please don't add all data so it will be easier to read, here it's not readable, to make it readable you need to use example request JSON before showing the CTP representation. 🙏

File: extension/docs/WebComponentsIntegrationGuide.md (hunk 2 above)
Comment (lojzatran) [reply extension/docs/WebComponentsIntegrationGuide.md:178]: We could, but their example does not reflect reality, that's why it is so short.  When you're trying to make payment, you usually need to provide much more info. Their example doesn't work with credit cards nor Klarna, so I didn't want to provide a nice, but non functioning example.

File: extension/docs/WebComponentsIntegrationGuide.md (hunk 3 above)
Comment (lojzatran) [reply extension/docs/WebComponentsIntegrationGuide.md:155]: See comment above: https://github.com/commercetools/commercetools-adyen-integration/pull/335#discussion_r442759463. I was never be able to make payment with the example, so that's why I didn't want to put sth which does not reflect reality.

File: extension/docs/WebComponentsIntegrationGuide.md (hunk 2 above)
Comment (ahmetoz) [reply extension/docs/WebComponentsIntegrationGuide.md:178]: Not sure about the functionality part, the examples should have required data with good examples t, here please use data to show something it's not just something to run. Linking the same example with Adyen has advantages for linking them in a nice way. Because we are referring their documentations, so no context switch for data representations,  So Adyen documentation and their example data are useful also they  probably spend time for representations, someone will read those documents and think about it, it's not obvious for you because you are not focusing to understand.

TLDR; please spend time for the representations, for me it's the most important thing in the docs.

Do you think we can not improve the data below?

"https://9944328d3d5e.ngrok.io/return-
"paymentData\":\"xxx\"
"userAgent\":\"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.129 Safari/537.36\
"holderName\":\"Lam Tran\"

Comment (lojzatran) [reply extension/docs/WebComponentsIntegrationGuide.md]: I replaced it with more simple example. It's not the same as in Adyen docs as their example is a non-functional example. https://github.com/commercetools/commercetools-adyen-integration/pull/335/commits/05cfc52eaa423642beb881fe874bcd33856cf155

File: extension/docs/WebComponentsIntegrationGuide.md (hunk 1 above)
Comment (lojzatran) [reply extension/docs/WebComponentsIntegrationGuide.md:172]: https://github.com/commercetools/commercetools-adyen-integration/pull/335/commits/05cfc52eaa423642beb881fe874bcd33856cf155

File: extension/test/e2e/e2e-test-utils.js [hunk 4]
```diff
@@ -24,7 +24,7 @@ async function executeInAdyenIframe (page, selector, executeFn) {
 
 async function initPuppeteerBrowser () {
   return puppeteer.launch({
-    headless: true,
+    headless: false,
```
Comment (ahmetoz) [inline extension/test/e2e/e2e-test-utils.js:27]: it should be headless :D

Comment (lojzatran) [reply extension/test/e2e/e2e-test-utils.js:27]: Oh damn, good catch 😄

Comment (lojzatran) [reply extension/test/e2e/e2e-test-utils.js]: https://github.com/commercetools/commercetools-adyen-integration/pull/335/commits/faa5b3d639e628e02ef7dedcf84ee55f5ff9a09e

File: extension/docs/DeploymentGuide.md [hunk 5]
```diff
@@ -1,3 +1,18 @@
...
+    - [AWS Lambda](#aws-lambda)
+    - [Docker](#docker)
+      - [Pull the image](#pull-the-image)
+      - [Run the container](#run-the-container)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
 # Deployment Guide
 
 **Contents**
```
Comment (ahmetoz) [inline extension/docs/DeploymentGuide.md:18]: We have contents already in here, same for Development Guide..
